### PR TITLE
satellite bias coefficient conversion update including upgrader

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

provide updates to the satellite bias correction converter as well as an upgrade capability for existing files

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -361,7 +361,7 @@ if( iodaconv_bufr_ENABLED )
   )
 endif()
 
-if (iodaconv_satbias_ENABLED )
+if (iodaconv_varbc_gsi_ENABLED )
```
Comment (BenjaminRuston) [inline test/CMakeLists.txt:364]: understand the GSI name here as it is specific to your predictor choices, but maybe NOAA ?

**Comment by BenjaminRuston** [comment]
passed all the ctests:
```
  Test  #15: iodaconv_satbias_py_coding_norms
  Test  #16: iodaconv_satbias_coding_norms
  Test  #67: test_iodaconv_satbias_viirs_aod
  Test #139: test_iodaconv_satbias_amsua
  Test #140: test_iodaconv_satbias_cris
  Test #141: test_iodaconv_satbias_gmi
  Test #142: test_iodaconv_satbias_ssmis
```
locally would like to try on a few more files in the morning and discuss but overall seems like this too is ready

File: src/varbc_gsi/satbias_upgrader.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        bias_coeff = oldnc.variables['bias_coefficients'][:]
+        for i, pred in enumerate(predictors):
+            var1_out = newnc.createVariable(f"biasCoefficients/{pred}", "f4", ("nrecs", "nvars"),
+                                            fill_value=-3.36879526e+38)
+            var1_out[0, :] = bias_coeff[i, :]
+    if 'bias_coeff_errors' in oldnc.variables.keys():
+        bias_coeff_err = oldnc.variables['bias_coeff_errors'][:]
+        for i, pred in enumerate(predictors):
+            var2_out = newnc.createVariable(f"biasCoeffErrors/{pred}", "f4", ("nrecs", "nvars"),
+                                            fill_value=-3.36879526e+38)
```
Comment (BenjaminRuston) [inline src/varbc_gsi/satbias_upgrader.py:59]: definitely a wee-niggle but when I look at the:
`float_missing_value = iconv.get_default_fill_val('float32') = 9.969209968386869e+36`

**Comment by BenjaminRuston** [comment]
tested on a couple bias files `atms_n20` and `iasi_mtb` and all is working fine

**Review by shlyaeva — COMMENTED** [review]
I didn't test, and only looked through the code, changes look reasonable to me.

File: src/compo/viirs_biaswriter.py [hunk 3]
```diff
@@ -19,25 +19,18 @@
...
-coef_dim = ncfile.createDimension('bias_coefficients', 1)
-coef_err_dim = ncfile.createDimension('bias_coeff_errors', 1)
-npredictors = ncfile.createDimension('npredictors', 1)
-nchannels = ncfile.createDimension('nchannels', 1)
-coef = ncfile.createVariable('bias_coefficients', np.float, ('bias_coefficients', ))
-coef_error = ncfile.createVariable('bias_coeff_errors', np.float, ('bias_coeff_errors', ))
-predictors = ncfile.createVariable('predictors', 'S8', ('npredictors', ))
-channels = ncfile.createVariable('channels', np.int, ('nchannels', ))
-coef.units = 'Aerosol optical depth'
-coef.long_name = 'bias_coefficients'
```
Comment (huishao-r) [inline src/compo/viirs_biaswriter.py:34]: This brings a question to me whether we would be able to tell what variables these coefficients are for. It seems we are not leaving any trace in the coefficient files. Is it true?

**Comment by BenjaminRuston** [comment]
thanks @CoryMartin-NOAA testing again after merge of Kevin's PR

File: src/compo/viirs_biaswriter.py (hunk 3 above)
Comment (CoryMartin-NOAA) [reply src/compo/viirs_biaswriter.py:34]: There is a problem in that AOD and brightnessTemperature are ambiguous, we can know channel or variable but not both. Is that a problem? The obs space would never have Tb and AOD both would it?

**Comment by BenjaminRuston** [comment]
this is working for me @CoryMartin-NOAA please merge if you're ready or let me know something you'd like me to test.  I can't approve *my* PR

**Comment by CoryMartin-NOAA** [comment]
I'd like @ADCollard to give it a quick once-over but then I'm happy to merge it in.

**Comment by ADCollard** [comment]
The ctests all pass for me and I am happy with the changes.  I checked the radiance and aircraft output files and the contents appear as expected.

**Review by ADCollard — APPROVED** [review]

**Pull request merged**